torchbiggraph/converters/importers.py

```diff
@@ -178,24 +183,35 @@ def generate_entity_path_files(
     relation_types: Dictionary,
     dynamic_relations: bool,
 ) -> None:
-    print(f"Preparing counts and dictionaries for entities and relation types:")
+    log(f"Preparing counts and dictionaries for entities and relation types:")
     entity_storage.prepare()
     relation_type_storage.prepare()
 
     for entity_name, entities in entities_by_type.items():
         for part in range(entities.num_parts):
-            print(
+            log(
                 f"- Writing count of entity type {entity_name} " f"and partition {part}"
             )
             entity_storage.save_count(entity_name, part, entities.part_size(part))
             entity_storage.save_names(entity_name, part, entities.get_part_list(part))
 
     if dynamic_relations:
-        print("- Writing count of dynamic relations")
+        log("- Writing count of dynamic relations")
         relation_type_storage.save_count(relation_types.size())
         relation_type_storage.save_names(relation_types.get_list())
 
 
+def append_to_file(data, appender):
+    lhs_offsets, rhs_offsets, rel_ids = zip(*data)
```

Comment:
`x -> zip(*x)` does a transposition, right? Is it efficient to do it like this in Python? Could we have PyTorch do it more efficiently?

Reply:
Everything in this code is outrageously slow, this zip is fine in comparison.

The slow part before was literally creating Tensors, TensorLists, EdgeLists, EntityLists, and then calling some hdf5 appender stuff *for every single edge*.

Reply:
> `x -> zip(*x)` does a transposition, right? Is it efficient to do it like this in Python? Could we have PyTorch do it more efficiently?

Yes, when I modified it to pytorch's transposition, this **once again** speeds up graph import by 5-20x. Reference #198 .